crates/egui-wgpu/src/egui.wgsl

```diff
@@ -78,7 +101,11 @@ fn fs_main_linear_framebuffer(in: VertexOutput) -> @location(0) vec4<f32> {
     let tex_linear = textureSample(r_tex_color, r_tex_sampler, in.tex_coord);
     let tex_gamma = gamma_from_linear_rgba(tex_linear);
     let out_color_gamma = in.color * tex_gamma;
-    return vec4<f32>(linear_from_gamma_rgb(out_color_gamma.rgb), out_color_gamma.a);
+    let out_color_linear = linear_from_gamma_rgb(out_color_gamma.rgb);
+    // Dither the float color down to eight bits to reduce banding.
+    // This step is optional for egui backends.
+    let out_color_dithered = dither_interleaved(out_color_linear, 256.0, in.position);
```

Comment:
I agree, this should be done on the gamma color. It's important to know the context of when this fragment shader is used: https://github.com/emilk/egui/blob/ff7a3832b68a8ecccf531511d1a17d92821b5430/crates/egui-wgpu/src/renderer.rs#L319 it is enabled in the case of using an srgb converting target. That means whatever float value we emit here will be _upon write_ converted to an srgb value. I.e. as-if the output of this method was wrapped in another call of `gamma_from_linear_rgba`. 

Therefore, we have to make sure that all the meaningful value changes happen on the gamma color.

Note that it would be an entirely different story if we're outputing _anything_ that isn't ending up on a `wgpu::Surface` that is targeting a srgb interpreting monitor/window. Meaning that all of this breaks down for HDR.

Reply:
I assumed that the output buffer was linear in that case dithering the linear color is what is wanted. If it's converted back to srgb before storage, it isn't optimal. I'll need to double check that.

Reply:
Just saw the comment be @Wumpf bellow, so you are right, this is off.